src/safeposix/syscalls/net_calls.rs

```diff
@@ -337,27 +406,40 @@ impl Cage {
         0
     }
 
+    //Helper function of bind_syscall
+    //Task: checks if fd refers to a valid socket file descriptor
+    //fd: the file descriptor associated with the socket
+    //localaddr: reference to the GenSockaddr enum that hold the address
+    //prereserved: bool that describes whether the address and port have 
+    //             been set aside or designated for specific purposes
     pub fn bind_inner(
         &self,
         fd: i32,
         localaddr: &interface::GenSockaddr,
         prereserved: bool,
     ) -> i32 {
+        //chekedfd is an atomic reference count of a lock on the fd
```

Comment:
```suggestion
        //checkedfd is an atomic reference count of a lock on the fd
```

Of a lock?  Or of the number of locks?  Or what?